src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs

```diff
@@ -1,119 +1,78 @@
-                                break;
-                            default:
-                                break;
-                        }
-                    }
-                    else
-                    {
-                        switch (timeProperty)
-                        {
-                            case TimeProperty.CreationTime:
-                                dir.CreationTime = dt;
-                                break;
-                            case TimeProperty.LastAccessTime:
-                                dir.LastAccessTime = dt;
-                                break;
-                            case TimeProperty.LastWriteTime:
-                                dir.LastWriteTime = dt;
-                                break;
-                            default:
-                                break;
-                        }
-                    }
+                DateTime dt = new DateTime(2014, 12, 1, 12, 0, 0, tuple.Item3);
+                tuple.Item1(testDir, dt);
+                Assert.Equal(dt, tuple.Item2(testDir));
+            });
+        }
 
-                    DateTime actual, actualUtc;
-                    switch (timeProperty)
-                    {
-                        case TimeProperty.CreationTime:
-                            actual = dir.CreationTime;
-                            actualUtc = dir.CreationTimeUtc;
-                            break;
-                        case TimeProperty.LastAccessTime:
-                            actual = dir.LastAccessTime;
-                            actualUtc = dir.LastAccessTimeUtc;
-                            break;
-                        case TimeProperty.LastWriteTime:
-                            actual = dir.LastWriteTime;
-                            actualUtc = dir.LastWriteTimeUtc;
-                            break;
-                        default:
-                            throw new ArgumentException("Invalid time property type");
-                    }
+        [Fact]
+        public void CreationSetsAllTimes()
+        {
+            string path = GetTestFilePath();
+            long beforeTime = DateTime.Now.AddSeconds(-3).Ticks;
+            long utcBeforeTime = DateTime.UtcNow.AddSeconds(-3).Ticks;
 
-                    DateTime expected = dt.ToLocalTime();
-                    DateTime expectedUtc = dt.ToUniversalTime();
+            DirectoryInfo testDirectory = new DirectoryInfo(GetTestFilePath());
+            testDirectory.Create();
 
-                    if (dt.Kind == DateTimeKind.Unspecified)
-                    {
-                        if (setUtc)
-                        {
-                            expectedUtc = dt;
-                        }
-                        else
-                        {
-                            expected = dt;
-                        }
-                    }
+            long afterTime = DateTime.Now.AddSeconds(3).Ticks;
```

Comment:
Ideally I think we'd only capture the UTC times, and then below rather than choosing whether to compare against the local or UTC, we instead decide whether or not to convert the provided time to UTC based on whether it already is.  Math against local times is usually something we try to avoid.